client/src/components/chat/chat.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Created by einavcarmon on 19/06/2017.
+ */
+
+import React, { Component } from 'react';
+import { createChat } from '../../api';
+import { Tabs, Tab } from 'react-bootstrap';
+import io from 'socket.io-client'
+
+import { Modal, Button } from 'react-bootstrap'
+import Strings from '../../strings'
+import { BackToFrequentBtn } from '../common';
+
+import ChatTab from "./chatTab";
+import './chat.css'
+
+export default class chatTab extends Component {
+
+    constructor (props) {
+        super(props);
+
+        this.state = {
+            chats : [],
+            client: props.user ? props.user.firstName : '',
+            activeChatIndex: 0,
+            showModal:true,
+            user: props.user
+        };
+
+        // todo move url to .env (@dancar help ?)
+        // this.socket = io.connect(`http://${process.env.REACT_APP_CHAT_HOST}:${process.env.REACT_APP_CHAT_PORT}`);
+        this.socket = io.connect(`http://localhost:8080`);
+        // this.socket = io.connect(`http://10.0.0.4:8080`);
+    }
+
+    componentWillReceiveProps (nextProps){
+        // b/c we read the user from the cookie it takes time for it to
+        // be set
+        this.setState({user:nextProps.user});
+        if (this.state.user)
+        {
+            this.setState({client:this.state.user.firstName});
+        }
+    }
+
+    closeModal() {
+        this.setState({ showModal: false });
+    }
+
+    createChat(){
+        var self = this;
+
+        createChat(this.state.client)
+        .then( (resp) =>  {
+            self.setState({
+                chats: self.state.chats.concat(resp)
+            })
+        })
+        .catch(function (err){
+            // todo what's here?
+        })
+    }
+
+    componentWillMount(){
+        if (this.state.user){
+            this.setState({client:this.state.user.firstName});
+            this.createChat();
+        }
+    }
+
+    handleChatChange (index) {
+        this.setState({activeChatIndex : index });
+    }
+
+    render() {
+        return (
+            <div className="all-over">
+                <Modal show={!this.state.user && this.state.showModal} onExited = {() => this.createChat()}>
```

Comment:
yeah that's a major issue, that's why I was having so much trouble with getting the user aftr the modal has been mounted, a thing I can assume can be solved with redux.